// aws-amplify-angular/dist/src/components/authenticator/sign-in-component/sign-in.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { includes } from '../common';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-container\" data-test=\"" + auth.signIn.section + "\">\n    <div class=\"amplify-form-body\" data-test=\"" + auth.signIn.bodySection + "\">\n      <div class=\"amplify-form-header\" data-test=\"" + auth.signIn.headerSection + "\">\n        {{ this.amplifyService.i18n().get('Sign in to your account') }}\n      </div>\n      <div class=\"amplify-amplify-form-row amplify-signin-username\">\n        <label class=\"amplify-input-label\" for=\"amplifyUsername\">\n          {{ this.amplifyService.i18n().get('Username *') }}\n        </label>\n        <input\n          #amplifyUsername\n          (keyup)=\"setUsername($event.target.value)\"\n          class=\"amplify-form-input\"\n          type=\"text\"\n          required\n          placeholder=\"{{ this.amplifyService.i18n().get('Username') }}\"\n          [value]=\"username\"\n          data-test=\"" + auth.signIn.usernameInput + "\"\n        />\n      </div>\n      <div class=\"amplify-form-row amplify-signin-password\">\n        <label class=\"amplify-input-label\" for=\"password\">\n          {{ this.amplifyService.i18n().get('Password *') }}\n        </label>\n        <input #password\n          (keyup)=\"setPassword(password.value)\"\n          (keyup.enter)=\"onSignIn()\"\n          class=\"amplify-form-input\"\n          type=\"password\"\n          required\n          placeholder=\"{{ this.amplifyService.i18n().get('Enter your password') }}\"\n          data-test=\"" + auth.signIn.passwordInput + "\"\n        />\n        <span class=\"amplify-form-action\" *ngIf=\"!shouldHide('ForgotPassword')\">{{ this.amplifyService.i18n().get('Forgot Password?') }}\n        <a class=\"amplify-form-link\"\n            (click)=\"onForgotPassword()\"\n            data-test=\"" + auth.signIn.forgotPasswordLink + "\"\n          >{{ this.amplifyService.i18n().get('Reset your password') }}</a></span>\n      </div>\n      <div class=\"amplify-form-actions\">\n        <div class=\"amplify-form-cell-right\">\n          <button class=\"amplify-form-button\"\n            (click)=\"onSignIn()\"\n            data-test=\"" + auth.signIn.signInButton + "\"\n          >{{ this.amplifyService.i18n().get('Sign In') }}</button>\n        </div>\n        <div class=\"amplify-form-cell-left\" *ngIf=\"!shouldHide('SignUp')\">\n          <div class=\"amplify-form-signup\">\n            {{ this.amplifyService.i18n().get('No account?') }}\n            <a\n              class=\"amplify-form-link\"\n              (click)=\"onSignUp()\"\n              data-test=\"" + auth.signIn.createAccountLink + "\"\n              >\n              {{ this.amplifyService.i18n().get('Create account') }}\n            </a>\n          </div>\n        </div>\n      </div>\n    </div>\n  </div>\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div\n        class=\"amplify-alert-message\"\n        data-test=\"" + auth.signIn.signInError + "\"\n        >\n          {{ this.amplifyService.i18n().get(errorMessage) }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n</div>\n";
var SignInComponentCore = /** @class */ (function () {
    function SignInComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
        this.hide = [];
        this.logger = this.amplifyService.logger('SignInComponent');
    }
    Object.defineProperty(SignInComponentCore.prototype, "data", {
        set: function (data) {
            this.hide = data.hide ? data.hide : this.hide;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SignInComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = includes(['signIn', 'signedOut', 'signedUp'], authState.state);
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    SignInComponentCore.prototype.ngOnInit = function () {
        if (!this.amplifyService.auth()) {
            throw new Error('Auth module not registered on AmplifyService provider');
        }
    };
    SignInComponentCore.prototype.shouldHide = function (comp) {
        return this.hide.filter(function (item) { return item === comp; })
            .length > 0;
    };
    SignInComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    SignInComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    SignInComponentCore.prototype.onSignIn = function () {
        var _this = this;
        this.amplifyService.auth().signIn(this.username, this.password)
            .then(function (user) {
            if (user['challengeName'] === 'SMS_MFA' || user['challengeName'] === 'SOFTWARE_TOKEN_MFA') {
                _this.amplifyService.setAuthState({ state: 'confirmSignIn', user: user });
            }
            else if (user['challengeName'] === 'NEW_PASSWORD_REQUIRED') {
                _this.amplifyService.setAuthState({ state: 'requireNewPassword', user: user });
            }
            else {
                _this.amplifyService.setAuthState({ state: 'signedIn', user: user });
            }
        })
            .catch(function (err) {
            _this._setError(err);
        });
    };
    SignInComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    SignInComponentCore.prototype.onForgotPassword = function () {
        this.onAlertClose();
        var user = this.username ? { username: this.username } : null;
        this.amplifyService.setAuthState({ state: 'forgotPassword', user: user });
    };
    SignInComponentCore.prototype.onSignUp = function () {
        this.onAlertClose();
        var user = this.username ? { username: this.username } : null;
        this.amplifyService.setAuthState({ state: 'signUp', user: user });
    };
    SignInComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
        this.logger.error(this.errorMessage);
    };
    SignInComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-in-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    SignInComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
    SignInComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "hide": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return SignInComponentCore;
}());
export { SignInComponentCore };
